sklearn/ensemble/tests/test_weight_boosting.py

```diff
@@ -630,3 +630,21 @@ def test_base_estimator_property_deprecated(AdaBoost):
     )
     with pytest.warns(FutureWarning, match=warn_msg):
         model.base_estimator_
+
+
+# TODO(1.4): remove in 1.4
+def test_deprecated_base_estimator_parameters_can_be_set():
+    """Check that setting base_estimator parameters works.
+
+    During the deprecation cycle setting "base_estimator__*" params should
+    work.
+
+    Non-regression test for https://github.com/scikit-learn/scikit-learn/issues/25470
+    """
+    # This implicitly sets "estimator", it is how old code (pre v1.2) would
+    # have instantiated AdaBoostClassifier and back then it would set
+    # "base_estimator".
+    clf = AdaBoostClassifier(DecisionTreeClassifier())
```

Comment:
actually here we are setting `estimator` and not `base_estimator`. It is completely fine that it works. However, could we parametrize the non-regression test to check that both works when setting `estimator` and `base_estimator`.

It could also be work to call `fit` just to make sure that there is not latter side-effect.

Reply:
I think this test is enough. If `clf = AdaBoostClassifier(base_estimator=DecisionTreeClassifier())` then I would expect that the warning would be raised at `__init__` time and no warning would be raised when calling `clf.set_params(base_estimator__max_depth=2)`.

Reply:
This is indeed what happens. I also think the warning at `fit` time is enough in this case.